From 7b580e465eff18991077dab4c4a293a9e57b4180 Mon Sep 17 00:00:00 2001
From: David Mabe <dmmabe@ncsu.edu>
Date: Sun, 29 Mar 2015 19:12:24 -0400
Subject: [PATCH] Adapting library's install_name on Darwin to installation 
 prefix.

Patch from Robin Sommer <robin@icir.org> re-rolled to LLVM 3.6.
---
 lib/CMakeLists.txt | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index c0ca9d9..9b6bd16 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -66,8 +66,8 @@ if ( APPLE )
     list(APPEND LIBCXX_LINK_FLAGS
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
-      "-Wl,-reexport_library,/usr/lib/libc++abi.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
+      "-Wl,-reexport_library,${CMAKE_INSTALL_PREFIX}/lib/libc++abi.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "/usr/lib/libSystem.B.dylib")
   else()
@@ -78,8 +78,8 @@ if ( APPLE )
           "${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib"
           "-Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++sjlj-abi.exp")
       else()
-        set(OSX_RE_EXPORT_LINE
-          "-Wl,-reexport_library,${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib")
+        get_property(abiloc TARGET cxxabi PROPERTY LOCATION)
+        set(OSX_RE_EXPORT_LINE "-Wl,-reexport_library,${abiloc}")
       endif()
     else()
       set (OSX_RE_EXPORT_LINE "/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++abi${LIBCXX_LIBCPPABI_VERSION}.exp")
@@ -87,11 +87,12 @@ if ( APPLE )
 
     list(APPEND LIBCXX_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "${OSX_RE_EXPORT_LINE}"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/notweak.exp"
-      "-Wl,-force_symbols_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/weak.exp")
+      "-Wl,-force_symbols_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/weak.exp"
+      "-nodefaultlibs")
   endif()
 endif()
 
@@ -105,6 +106,7 @@ set_target_properties(cxx
     OUTPUT_NAME   "c++"
     VERSION       "1.0"
     SOVERSION     "1"
+    INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
   )
 
 install(TARGETS cxx
-- 
1.9.5 (Apple Git-50.3)

